### DISCUSSION TITLE
Importing AggregatorV3Interface issues

### DISCUSSION THREAD

**Discussion by leopashov** [smartcontractkit/full-blockchain-solidity-course-py#121] [Q&A] [answered]
I'm running into trouble importing the AggregatorV3Interface.

Attempting the original method suggested - using the @chainlink remapping I get:

`CompilerError: solc returned the following errors:

contracts/Lottery.sol:5:1: ParserError: Source "C:/Users/leopa/.brownie/packages/smartcontractkit/chainlink-brownie-contracts@1.1.1/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol" not found: File not found.`

when trying to compile.
I then tried just typing the path in full: 
`import "smartcontractkit/chainlink-brownie-contracts@1.1.1/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";`

which got me: 

`CompilerError: solc returned the following errors:

contracts/Lottery.sol:7:1: ParserError: Source "smartcontractkit/chainlink-brownie-contracts@1.1.1/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol" not found: File outside of allowed directories.
import "smartcontractkit/chainlink-brownie-contracts@1.1.1/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

I got the same with the full github link.

I then tried copying and pasting the AggragtorV3Interface.sol code into a new AggregatorV3Interface.sol file in my local directory and importing with:

`import "./AggregatorV3Interface.sol";`

But I get:

`CompilerError: solc returned the following errors:

contracts/AggregatorV3Interface.sol:3:1: ParserError: Source file requires different compiler version (current compiler is 0.6.12+commit.27d51765.Windows.msvc) - note that nightly builds are considered to be strictly less than the released version
pragma solidity ^0.8.0;`

From https://ethereum.stackexchange.com/questions/102470/brownie-compile-using-different-compiler-version-than-specified-in-configurati I guess this is because the solidity versions are different and I'm trying to import one into the other but isn't this also the case in the original method?

Deleting the `pragma solidity ^0.8.0;` line from the AggregatorV3Interface file allows it to compile but will this lead to issues down the line?

Any help would be greatly appreciated.

Also I'm on windows if that makes any difference.

Thank you

**Comment by davidyuill27** [comment]
Might need more info but maybe try changing the v0.8 in the import string to v0.6 as I think most of the projects Patrick done are in Solidity major version 0.6.6

**Comment by AlbertoSinigaglia** [comment]
You might have installed the wrong version of the solidity compiler, check https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/5

**Comment by PatrickAlphaC** [comment, marked as answer]
Looks like you’re trying to import v0.8 instead of v0.6. Version `1.1.1` of the brownie contracts packages doesn’t have v0.8, `0.2.2` does tho. So you have two option:

1. Have everything use `0.6` editions
2. Use  `import "smartcontractkit/chainlink-brownie-contracts@0.2.2/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";`

remember, you’ll need to update your dependencies and remapping in your config with this! 

I know it looks like you’re using an “earlier” version of the contracts, but after `1.2.0` the versioning was changed to match those in npm, so `0.2.2` is actually currently the most up to date. 

Let me know if this helps

**Reply by ChristianG77** [reply]
Hi Patrick.

Thanks for the course this is amazing!

I've input the following:

pragma solidity ^0.6.6;

import "smartcontractkit/chainlink-brownie-contracts@0.2.2/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
import "smartcontractkit/chainlink-brownie-contracts@0.2.2/contracts/src/v0.7/vendor/SafeMathChainlink.sol";

And I'm getting the following compile error message, 'File outside of allowed directories.'.

PS C:\Users\chris\Dropbox\00_CODE\brownie_fund_me> brownie compile
INFO: Could not find files for the given pattern(s).
Brownie v1.16.4 - Python development framework for Ethereum

Compiling contracts...
  Solc version: 0.6.12
  Optimizer: Enabled  Runs: 200
  EVM Version: Istanbul
CompilerError: solc returned the following errors:

contracts/FundMe.sol:5:1: ParserError: Source "smartcontractkit/chainlink-brownie-contracts@0.2.2/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol" not found: File outside of allowed directories.
import "smartcontractkit/chainlink-brownie-contracts@0.2.2/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
^------------------------------------------------------------------------------------------------------------------^

contracts/FundMe.sol:6:1: ParserError: Source "smartcontractkit/chainlink-brownie-contracts@0.2.2/contracts/src/v0.7/vendor/SafeMathChainlink.sol" not found: File outside of allowed directories.
import "smartcontractkit/chainlink-brownie-contracts@0.2.2/contracts/src/v0.7/vendor/SafeMathChainlink.sol";

Any help would be much appreciated.

Thanks.

**Reply by ChristianG77** [reply]
I got it to work using version 1.1.1.

Are there any issues with using this version versus 0.2.2?

You'd probably want to use the latest version, correct?

**Reply by PatrickAlphaC** [reply]
Perfect!

So... there are some oddities with the versioning actually. We wanted the chainlink-brownie contract versions to match the NPM versions, so we pushed back the brownie contracts after 1.2.0 to match npm contracts.

So 0.2.2 is actually the most up to date. Oddly enough.

I'm glad you got the contracts to work with 1.1.1 though!

**Reply by leopashov** [reply]
I downloaded the 1.1.1 release and used the V3 aggregator code which was written in version 0.6, all seems to be working so far, thank you very much for your help and for the course!
